brave/brave-ios
# Fix #6851: Move IPFS settings under Web3 settings

## Summary of Changes
Move IPFS settings code from Client to Wallet 
Rewrote IPFS settings in SwiftUI 
Move IPFS settings UI under Web3 


This pull request fixes #6851

## Submitter Checklist:


## Test Plan:
1. check there is no IPFS under Settings 
2. check there is an IPFS section under Web3
3. check if one can set a custom NFT gateway url 
    - ex. `https://nftstorage.link` /  `https://dweb.link`

## Screenshots:


![simulator_screenshot_56923D2F-DF7C-4B91-9F45-2004F81BF291](https://user-images.githubusercontent.com/1187676/216748417-d2dd2315-148b-41d0-a703-d19c11b1461a.png) | ![simulator_screenshot_1F28BF9B-EA0F-4787-88D7-824507583614](https://user-images.githubusercontent.com/1187676/216748419-e3f332f2-c906-4adb-aa17-f9d4e8730abb.png) | ![simulator_screenshot_2BBB266C-F1C6-4EBC-B758-9F7873FF7DC1](https://user-images.githubusercontent.com/1187676/216748423-9545f16f-9804-4c0b-8db9-00c6b7fa6de2.png)
--|--|--

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
